iceoryx_hoofs/include/iceoryx_hoofs/cxx/helplets.hpp

```diff
@@ -338,7 +338,14 @@ bool doesEndWithPathSeparator(const string<StringCapacity>& name) noexcept;
 /// @param[in] value of type F to convert to T
 /// @return converted value of F to corresponding value of T
 template <typename F, typename T>
```

Comment:
1. I would rename the template arguments `F` and `T` to `SourceType` and `DestinationType` which would it make much more readable.
2. Furthermore I would switch the template parameter order which would allow the user to use this function by providing only one template argument.
3. Could you please rename `from` into `into`. In Rust it makes sense to have a `From` trait since you can add additional methods to already pre-existing types - a feature C++ does not have. Just look at this code usage example (with the already reverted template parameter order):
```cxx
int destination = from<int>(12.12f);
```
This reads weird, it is not from `int`. The float `12.12f` is converted into `int` therefore the name change.

So my final solution would look like this:
```cxx
template <typename SourceType, typename DestinationType>
SourceType into(const DestinationType& value) {}
```